main/src/main/scala/sbt/Defaults.scala

```diff
@@ -1625,6 +1634,28 @@ object Defaults extends BuildCommon {
           else old
         }
       ))
+
+  private case class JavaDiscoverConf(root: String, dirRexp: Regex, innerDir: String)
+  private final val JavaDiscoverConfigs = Seq(
+    JavaDiscoverConf("/usr/lib/jvm", "java-([0-9]+)-.*".r, ""), // most linux
+    JavaDiscoverConf("/Library/Java/JavaVirtualMachines",
+                     "jdk[1\\.]*([0-9]+)\\..*".r,
```

Comment:
Doesn't this match, for instance, `/Library/Java/JavaVirtualMachines/jdk-10.jdk/`?

Reply:
Good catch. I'll change it to `jdk(1\.|-)?([0-9]+)\..*` and extract the second group.